langstream-agents/langstream-agent-azure-blob-storage-source/src/main/java/ai/langstream/agents/azureblobstorage/source/AzureBlobStorageSource.java

```diff
@@ -0,0 +1,260 @@
+                log.debug("Skipping blob with bad extension {}", name);
+                continue;
+            }
+            if (!blobsToCommit.contains(name)) {
+                log.info("Found new blob {}", name);
+                try {
+                    byte[] read = client.getBlobClient(name).downloadContent().toBytes();
+                    ;
+                    records.add(new BlobSourceRecord(read, name));
+                    somethingFound = true;
+                    blobsToCommit.add(name);
+                } catch (Exception e) {
+                    log.error("Error reading object {}", name, e);
+                    throw e;
+                }
+                break;
+            } else {
+                log.info("Skipping already processed object {}", name);
+            }
+        }
+        if (!somethingFound) {
+            log.info("Nothing found, sleeping for {} seconds", idleTime);
+            Thread.sleep(idleTime * 1000L);
+        } else {
+            processed(0, 1);
+        }
+        return records;
+    }
+
+    static boolean isExtensionAllowed(String name, Set<String> extensions) {
+        if (extensions.contains(ALL_FILES)) {
+            return true;
+        }
+        String extension;
+        int extensionIndex = name.lastIndexOf('.');
+        if (extensionIndex < 0 || extensionIndex == name.length() - 1) {
+            extension = "";
+        } else {
+            extension = name.substring(extensionIndex + 1);
+        }
+        return extensions.contains(extension);
+    }
+
+    @Override
+    protected Map<String, Object> buildAdditionalInfo() {
+        return Map.of("container", client.getBlobContainerName());
+    }
+
+    @Override
+    public void commit(List<Record> records) throws Exception {
+        for (Record record : records) {
+            BlobSourceRecord blobRecord = (BlobSourceRecord) record;
+            String name = blobRecord.name;
+            log.info("Removing blob {}", name);
+            client.getBlobClient(name).deleteIfExists();
+            blobsToCommit.remove(name);
+        }
+    }
+
+    private static class BlobSourceRecord implements Record {
+        private final byte[] read;
+        private final String name;
+
+        public BlobSourceRecord(byte[] read, String name) {
+            this.read = read;
+            this.name = name;
+        }
+
+        /**
+         * the key is used for routing, so it is better to set it to something meaningful. In case
+         * of retransmission the message will be sent to the same partition.
+         *
+         * @return the key
+         */
+        @Override
+        public Object key() {
+            return name;
+        }
+
+        @Override
+        public Object value() {
+            return read;
+        }
+
+        @Override
+        public String origin() {
+            return null;
+        }
+
+        @Override
+        public Long timestamp() {
+            return System.currentTimeMillis();
```

Comment:
This should be set at instantiation time, otherwise different calls with return different results

Reply:
interesting, I copied it from S3 source